File: c++/pointers/address_access.cpp
/*
 * Copyright (C) 2019-2026 Katarzyna Krasińska
 * Languagium - https://github.com/katheroine/languagium
 * Licensed under GPL-3.0 - see LICENSE.md
 */
// SPDX-License-Identifier: GPL-3.0-or-later

#include <iostream>

int main() {
  int address = 1024;
  short int *pointer;

  pointer = reinterpret_cast<short int *>(address);

  // std::cout << *pointer << std::endl; // RUNTIME: Segmentation fault
}

File: pascal/pointers/address_access.pas
(*
 * Copyright (C) 2019-2026 Katarzyna Krasińska
 * Languagium - https://github.com/katheroine/languagium
 * Licensed under GPL-3.0 - see LICENSE.md
 *)

program AddressAccess;

type
  intptr = ^integer;

var
  address: integer = 1024;
  pointer: ^integer;

begin
  pointer := intptr(address);

  // writeln(pointer^); // Runtime error
end.
